### PR TITLE
Add multi_dense_vector value access to scripts

### PR DESCRIPTION
**Pull request by benwtrent**
This adds value access to multi_dense_vector values in scripts. The users will get:

 - Count of vectors per field
 - Magnitudes of all the individual vectors
 - Access to each vector with an iterator

I will happily take design critiques around how these are exposed in scripting.

I initially though of just providing directly `float[][]` access, but this seems to have some unfavorable behavior around creating a TON of garbage. The reason is that each field could have a different number of vectors, so allocating a new collection of `float[dim]` for every field seemed rough. 

Generally, when scripting or using the vectors, an iterator should be enough and I have the iterator backed by a simple buffer to keep garbage down.

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValues.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     * Get dense vector's magnitude
+     */
+    public float[] getMagnitudes() {
+        return getCheckedVector().getMagnitudes();
+    }
+
+    @Override
+    public BytesRef get(int index) {
+        throw new UnsupportedOperationException(
+            "accessing a multi-vector field's value through 'get' or 'value' is not supported, use 'vectorValues' or 'magnitudes' instead."
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorScriptDocValues.java:60]: It is different how script values on other fields operate: if we provide size(), there also an expectation to get each individual element through `get`.

But I understand a reasoning behind providing an Iterator. And also we don't expect users to access these values directly in script; ideally values should be accessed through functions.

**Review by mayya-sharipova — APPROVED** [review]
@benwtrent Thanks Ben, makes sense.

I am not very happy about API using Iterator, but I support reasoning for it.

**Review by jdconrad — APPROVED** [review]
LGTM! I do wish we had gotten farther in the deprecation of doc-style access so we didn't need `ScriptDocValues` anymore, but that's a completely different issue.

File: modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/181_multi_dense_vector_dv_fields_api.yml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      headers:
+        Content-Type: application/json
+      search:
+        rest_total_hits_as_int: true
+        body:
+          query:
+            script_score:
+              query: {match_all: {} }
+              script:
+                source: "doc['vector'].vectorValues.next()[0];"
```
[the remark this answers is not in the document]
Comment (jdconrad) [reply modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/181_multi_dense_vector_dv_fields_api.yml:130]: If this ever does become required, I could see the float[][] being lazy-loaded, but that would make the actual script execution so incredibly slow.

**Pull request merged**